File: SDKs/rudder.example.local/13/python/src/sdk/models/shared/cvecheck.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclass
class CveCheckPackages:
    r"""CveCheckPackages
    Packages affected by this CVE
    """
    
    name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    version: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('version') }})
    

@dataclass_json
@dataclass
class CveCheck:
    cve_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cveId') }})
    nodes: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('nodes') }})
    packages: Optional[List[CveCheckPackages]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('packages') }})
    

File: SDKs/rudder.example.local/13/typescript/src/sdk/models/shared/cvecheck.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// CveCheckPackages
/** 
 * Packages affected by this CVE
**/
export class CveCheckPackages extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=version" })
  version?: string;
}


export class CveCheck extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=cveId" })
  cveId?: string;

  @SpeakeasyMetadata({ data: "json, name=nodes" })
  nodes?: string[];

  @SpeakeasyMetadata({ data: "json, name=packages", elemType: CveCheckPackages })
  packages?: CveCheckPackages[];
}
